core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamUnitTest.java

```diff
@@ -0,0 +1,90 @@
+        Path sampleOut = tempDir.resolve("sample-out.txt");
+        List<String> lines = Arrays.asList("Hello. This is just a test. Good bye.");
+        Files.write(sampleOut, lines);
+        File sampleOutFile = sampleOut.toFile();
+        try (InputStream inputStream = new FileInputStream(sampleOutFile)) {
+            assertThat(readString(inputStream)).contains(lines.get(0));
+        }
+    }
+
+    @Test
+    public void givenAStringWrittenToFile_whenReadWithFileInputStreamWithStringConstructor_thenItShouldExitsInTheInputStream(@TempDir Path tempDir) throws IOException {
+        Path sampleOut = tempDir.resolve("sample-out.txt");
+        String expectedText = "Hello. Hi.";
+        List<String> lines = Arrays.asList(expectedText);
+        Files.write(sampleOut, lines);
+        String fileAbsolutePath = sampleOut.toFile()
+            .getAbsolutePath();
+        try (FileInputStream fis = new FileInputStream(fileAbsolutePath)) {
+            int content;
+            int availableBytes = fis.available();
+            Assert.assertTrue(availableBytes > 0);
+            StringBuilder actualReadText = new StringBuilder();
+            while ((content = fis.read()) != -1) {
+                actualReadText.append((char) content);
+            }
+            assertThat(actualReadText.toString()).contains(expectedText);
+        }
+    }
+
+    @Test
+    public void givenAByteArray_whenReadWithByteArrayInputStream_thenTheStringValueOfTheByteArrayShouldMatchTheExpectedString() throws IOException {
+        byte[] byteArray = { 104, 101, 108, 108, 111 };
+        try (ByteArrayInputStream bais = new ByteArrayInputStream(byteArray)) {
+            assertThat(readString(bais)).isEqualTo("hello");
+        }
+    }
+    @Test
+    public void givenKeyValuePairsWrittenInAFile_whenPassedInOutputStreams_thenThePairsShouldExistsInObjectInputStreams(@TempDir Path tempDir) throws IOException, ClassNotFoundException {
+        Path sampleOut = tempDir.resolve("sample-out.txt");
+        File textFile = sampleOut.toFile();
+        //Serialize a Hashmap then write it into a file.
+        Map<String, String> kv = new HashMap<>();
+        try (ObjectOutputStream objectOutStream = new ObjectOutputStream(new FileOutputStream(textFile))) {
+                kv.put("baseURL", "baeldung.com");
+                kv.put("apiKey", "this_is_a_test_key");
+                objectOutStream.writeObject(kv);
+        }
+        //Deserialize the contents of a file then transform it back into a Hashmap
+        try (ObjectInputStream input = new ObjectInputStream(new FileInputStream(textFile))) {
+                HashMap<String, String> inputKv = (HashMap<String, String>) input.readObject();
+                assertThat(kv.get("baseURL")).isEqualTo( inputKv.get("baseURL"));
+                assertThat(kv.get("apiKey")).isEqualTo( inputKv.get("apiKey"));
+        }
```

Comment:
Something weird is going on with the indentation here, please apply the Baeldung formatter

Reply:
Weird. I applied that formatter many times. I'll try again.